fair_mango/metrics/superset.py

```diff
@@ -0,0 +1,318 @@
+        by default None
+    positive_target : Sequence[int  |  float  |  str  |  bool] | None, optional
+        A Sequence of the positive labels corresponding to the provided
+        targets, by default None
+
+    Returns
+    -------
+    list
+        list
+        A list of dictionaries, each containing the sensitive attributes
+        considered and their corresponding fairness metric result.
+
+    Raises
+    ------
+    AttributeError
+        if data is a pandas dataframe and 'sensitive' parameter is not provided.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({
+    ...         'gender': ['male', 'male', 'male', 'female', 'female'],
+    ...         'race': ['white', 'black', 'black', 'white', 'white'],
+    ...         'real': [1,1,0,0,1],
+    ...         'pred': [0,1,0,0,1]
+    ... })
+    >>> result = super_set_fairness_metrics(
+    ...     metric=DemographicParityDifference,
+    ...     data=df,
+    ...     sensitive=['gender', 'race'],
+    ...     real_target=['real'],
+    ...     predicted_target=['pred'],
+    ... )
+    >>> result
+    [
+        {
+            'sensitive': ('gender',),
+            'result': {
+                'real': {
+                    ('male',): 0.16666666666666663,
+                    ('female',): -0.16666666666666663
+                }
+            }
+        },
+        {
+            'sensitive': ('race',),
+            'result': {
+                'real': {
+                    ('white',): 0.16666666666666663,
+                    ('black',): -0.16666666666666663
+                }
+            }
+        },
+        {
+            'sensitive': ('gender', 'race'),
+            'result': {
+                'real': {
+                    ('male', 'white'): 0.5,
+                    ('female', 'white'): -0.25,
+                    ('male', 'black'): -0.25
+                }
+            }
+        }
+    ]
+    """
+    results = []
+
+    if isinstance(data, Dataset):
+        sensitive = data.sensitive
+        real_target = data.real_target
+        predicted_target = data.predicted_target
+        positive_target = data.positive_target
+        data = data.df
+        if predicted_target == []:
+            predicted_target = None
+
+    if sensitive is None:
+        raise AttributeError(
+            "'sensitive' attribute is required when data is " "pandas dataframe"
+        )
+
+    pairs = chain.from_iterable(
+        combinations(sensitive, r) for r in range(1, len(sensitive) + 1)
+    )
+
+    for pair in pairs:
+        result = metric(
+            data=data,
+            sensitive=list(pair),
+            real_target=real_target,
+            predicted_target=predicted_target,
+            positive_target=positive_target,
+        ).rank()
+
+        results.append(
+            {
+                "sensitive": pair,
+                "result": result,
+            }
+        )
+
+    return results
+
+
+def super_set_performance_metrics(
```

Comment:
try to factorize it with the other superset function

Reply:
Yup, I created a function `_initialise` that initialises both of the superset classes.